fcd/ast/ast_context.cpp

```diff
@@ -331,7 +331,7 @@ class InstToExpr : public llvm::InstVisitor<InstToExpr, Expression*>
 					negated = right;
 				}
 			}
-			else if (auto constant = dyn_cast<ConstantInt>(inst.getOperand(1)))
+			else if ((constant = dyn_cast<ConstantInt>(inst.getOperand(1))))
```

Comment:
I actually think that this one is better off shadowing the variable that it replaces. I find it misleading to acknowledge that this value is replacing the one that was initially assigned in the `if` branch.